### ISSUE TITLE
Interfaces for the Process and ProcessModels of a Quadratic Programming (QP) solver on Loihi 2

### ISSUE THREAD

**Issue by ashishrao7** [lava-nc/lava-optimization#2]
This issue deals with creating an interfaces for unconstrained/ equality/ inequality constrained quadratic programming solvers for the form:

<img src="https://render.githubusercontent.com/render/math?math=\text{argmin}_x  \text{  } x^{T}Px %2B p^{T}x"> 
<img src="https://render.githubusercontent.com/render/math?math=\text{s.t.  }   Ax \ge b">

Here <img src="https://render.githubusercontent.com/render/math?math=P"> is a positive (semi)-definite matrix of  <img src="https://render.githubusercontent.com/render/math?math=\mathbb{R}^{n \times n}">.   <img src="https://render.githubusercontent.com/render/math?math=A"> is the matrix  in  <img src="https://render.githubusercontent.com/render/math?math=\mathbb{R}^{m \times n}"> describing  <img src="https://render.githubusercontent.com/render/math?math=m"> constraints for the vector <img src="https://render.githubusercontent.com/render/math?math=x \in \mathbb{R}^{n}">. 

The solver  would be part of the optimization toolbox for constrained optimization problems on Loihi.

**Comment by awintel** [comment]
Please set up the new directory structure within a new pull request and populate it with the class interfaces and a usage example.

**Comment by GaboFGuerra** [comment]
Will present the library design to @ashishrao7 and  @srrisbud in the morning, once I collect their feedback will pull request. Am adding you as assign for you to help in reviewing it.

**Comment by GaboFGuerra** [comment]
I'm opening the directory structure as a separate [issue #3](https://github.com/lava-nc/lava-optimization/issues/3). Because the library structure and QP solver interface are related but separate issues/discussions.

**Comment by GaboFGuerra** [comment]
Also added [issue 4](https://github.com/lava-nc/lava-optimization/issues/4), which could be started first (if not merged).

**Comment by ashishrao7** [comment]
Yes. I have been linking the PRs to the issue but often forget to close the issue :). Sorry about that.
